spring-kafka/src/test/java/org/springframework/kafka/listener/adapter/FilteringAdapterTests.java

```diff
@@ -72,4 +75,173 @@ public void testBatchFilterAckDiscard() throws Exception {
+				// SUT
+				return consumerRecords;
+			}
+
+			@Override
+			public boolean ignoreEmptyBatch() {
+				// SUT
+				return true;
+			}
+		};
+
+		final BatchAcknowledgingMessageListener<String, String> listener = mock(BatchAcknowledgingMessageListener.class);
+		final FilteringBatchMessageListenerAdapter<String, String> adapter =
+				new FilteringBatchMessageListenerAdapter<String, String>(listener, filter);
+		final List<ConsumerRecord<String, String>> consumerRecords = List.of(new ConsumerRecord<>("hello-topic", 1, 1, "hello-key", "hello-value"));
+		final Acknowledgment ack = mock(Acknowledgment.class);
+
+		final CountDownLatch latch = new CountDownLatch(1);
+		willAnswer(i -> {
+			latch.countDown();
+			return null;
+		}).given(listener).onMessage(any(List.class), any(Acknowledgment.class));
+
+		// When :
+		adapter.onMessage(consumerRecords, ack, null);
+
+		// Then
+		assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
+		verify(ack, never()).acknowledge();
+	}
+
+	@Test
+	public void listener_should_be_invoked_on_emptyList_and_ignoreEmptyBatch_false() throws Exception {
+		// Given :
+		final RecordFilterStrategy<String, String> filter = new RecordFilterStrategy<>() {
+			@Override
+			public boolean filter(ConsumerRecord<String, String> consumerRecord) {
+				return true;
+			}
+
+			@Override
+			public List<ConsumerRecord<String, String>> filterBatch(
+					List<ConsumerRecord<String, String>> consumerRecords) {
+				// SUT
+				return List.of();
+			}
+
+			@Override
+			public boolean ignoreEmptyBatch() {
+				// SUT
+				return false;
+			}
+		};
+
+		final BatchAcknowledgingMessageListener<String, String> listener = mock(BatchAcknowledgingMessageListener.class);
+		final FilteringBatchMessageListenerAdapter<String, String> adapter =
+				new FilteringBatchMessageListenerAdapter<String, String>(listener, filter);
+		final List<ConsumerRecord<String, String>> consumerRecords = new ArrayList<>();
+		final Acknowledgment ack = mock(Acknowledgment.class);
+
+		final CountDownLatch latch = new CountDownLatch(1);
+		willAnswer(i -> {
+			latch.countDown();
+			return null;
+		}).given(listener).onMessage(any(List.class), any(Acknowledgment.class));
+
+		// When :
+		adapter.onMessage(consumerRecords, ack, null);
+
+		// Then
+		assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
+		verify(ack, never()).acknowledge();
+	}
+
+	@Test
+	public void listener_should_be_invoked_on_notEmptyList_and_ignoreEmptyBatch_false() throws Exception {
+		// Given :
+		final RecordFilterStrategy<String, String> filter = new RecordFilterStrategy<>() {
+			@Override
+			public boolean filter(ConsumerRecord<String, String> consumerRecord) {
+
+				return true;
+			}
+
+			@Override
+			public List<ConsumerRecord<String, String>> filterBatch(
+					// SUT
+					List<ConsumerRecord<String, String>> consumerRecords) {
+				return consumerRecords;
+			}
+
+			@Override
+			public boolean ignoreEmptyBatch() {
+				// SUT
+				return false;
```

Comment:
DITTO

Reply:
Fixed it!

Reply:
Sure! Another request from me. No reason to say `fixed`, `done` etc. if you are agreed with me and will push that in the next commit or so. It is totally fine to argue if there is a disagreement or I am wrong. The point is that I have many other PRs to review or so and such a `Fixed` email is unnecessary distraction . 😅

Reply:
Oh, I'm really sorry. 😓 
I thought I had to do it that way...! 
Keep in my mind! i will left comment if it is needed. 

Sorry to bother you again 😅